src/hooks/FullRestrictions.sol

```diff
@@ -12,24 +12,32 @@ import {BaseHook} from "src/hooks/BaseHook.sol";
 contract FullRestrictions is BaseHook {
     constructor(address root_, address spoke_, address deployer) BaseHook(root_, spoke_, deployer) {}
 
-    /// @inheritdoc ITransferHook
-    function checkERC20Transfer(address from, address to, uint256, /* value */ HookData calldata hookData)
+    /// @inheritdoc BaseHook
+    function checkTransferPolicy(TransferType transferType, address from, address to, HookData calldata hookData)
         public
         view
         override
         returns (bool)
     {
-        if (isSourceOrTargetFrozen(from, to, hookData)) return false;
+        if (
+            transferType == TransferType.DepositFulfillment || transferType == TransferType.RedeemFulfillment
+                || transferType == TransferType.CrosschainTransfer
+        ) {
+            return true;
+        }
 
-        if (isDepositRequest(from, to)) return isTargetMember(to, hookData);
-        if (isDepositFulfillment(from, to)) return true;
-        if (isDepositClaim(from, to)) return isTargetMember(to, hookData);
-        if (isRedeemRequest(from, to)) return isSourceMember(from, hookData);
-        if (isRedeemFulfillment(from, to)) return true;
-        if (isRedeemClaim(from, to)) return isSourceMember(from, hookData);
-        if (isCrosschainTransfer(from, to)) return true;
+        if (
+            transferType == TransferType.DepositRequest || transferType == TransferType.DepositClaim
+                || transferType == TransferType.LocalTransfer
+        ) {
+            return isTargetMember(to, hookData);
+        }
 
-        // Else, it's a transfer
-        return isTargetMember(to, hookData);
+        if (transferType == TransferType.RedeemRequest || transferType == TransferType.RedeemClaim) {
+            return isSourceMember(from, hookData);
+        }
```

Comment:
Just syntactically and visually, I would prefer the line-by-line case we had (independently of using enum or not)